tool.ts

```diff
@@ -235,21 +233,42 @@ export async function downloadTool(
 
             tl.debug('downloading');
             let response: httpm.HttpClientResponse = await http.get(url, additionalHeaders);
-            
+
             if (response.message.statusCode != 200) {
                 let err: Error = new Error('Unexpected HTTP response: ' + response.message.statusCode);
                 err['httpStatusCode'] = response.message.statusCode;
                 tl.debug(`Failed to download "${fileName}" from "${url}". Code(${response.message.statusCode}) Message(${response.message.statusMessage})`);
                 throw err;
             }
 
+            let downloadedContentLength = _getContentLengthOfDownloadedFile(response);
+            if (!isNaN(downloadedContentLength)) {
+                tl.debug(`Content-Length of downloaded file: ${downloadedContentLength}`);
+            } else {
+                tl.debug(`Content-Length header missing`);
+            }
+
             tl.debug('creating stream');
             let file: NodeJS.WritableStream = fs.createWriteStream(destPath);
             file.on('open', async (fd) => {
                 try {
                     let stream = response.message.pipe(file);
                     stream.on('close', () => {
                         tl.debug('download complete');
+                        let fileSizeInBytes = _getFileSizeOnDisk(destPath);
+                        if (!isNaN(fileSizeInBytes)) {
```

Comment:
Could the file size from fs.stats result ever be NaN? Is this some real case?

Reply:
Actually it can not, but in case of any exception during fs operations I catch it and return NaN. So result of this function can be NaN

Reply:
Could it make sense to throw exception then and handle it while calling _getFileSizeOnDisk method? It's a little bit strange that function returns NaN in case of an error - instead of throwing an exception (it's not mentioned in the function description also and it could be unobvious to the customer). I believe exceptions mechanism is intended to handle such cases (NaN does not bring any info about error happened, and writing info inside of a function is against SOLID)

Reply:
Upd: since _getFileSizeOnDisk does not return NaN anymore - could it make sense to move  'Downloaded file size: ...'/'File size on disk was not found' messages to try..catch block above?